pkg/middleware/useragent/useragent.go

```diff
@@ -0,0 +1,22 @@
+package useragent
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	userAgentHeader string = "grpcgateway-user-agent"
+)
+
+// FromContext returns the user agent field stored in context.
+// If context does not have user agent field, function will return empty string and false.
+func FromContext(ctx context.Context) (string, bool) {
```

Comment:
If you move this into the `logging` middleware package, please rename to `UserAgentFromContext`.